common/eth2_network_config/src/lib.rs

```diff
@@ -207,20 +282,105 @@ impl Eth2NetworkConfig {
                         .map_err(|e| format!("Unable to read {:?}: {:?}", file, e))
                 })?;
 
-            Some(bytes).filter(|bytes| !bytes.is_empty())
+            let state = Some(bytes).filter(|bytes| !bytes.is_empty());
+            let genesis_state_source = if state.is_some() {
+                GenesisStateSource::IncludedBytes
+            } else {
+                GenesisStateSource::Unknown
+            };
+            (state, genesis_state_source)
         } else {
```

Comment:
It could be a nice feature to support `GenesisStateSource::Url` for custom testnets in the future - but may be an overkill since users have to download some other files anyway.

Reply:
Yeah it would be cool. We'd need to add a new file to define the URL, so I thought I'd leave that for future works.